ce/api/geo.py

```diff
@@ -106,13 +123,18 @@ def __call__(self, *args):
                     result = self.func(*args)
 
                 with cache_lock:
-                    self.cache[key] = result
+                    with cache_sizer(self.cache) as sizer:
+                        self.cache[key] = result
+                    self.size += getsize(result) + sizer.delta_size()
```

Comment:
OK, I learned something here. Python has a non-intuitive (to me) semantics for the scope of a variable defined in a `with` statement. It persists. Tricky. I guess my comment here is: this code is tricky. But I get what you are doing.

Reply:
Yeah, that definitely trips people up. There is no `with` scope.

Context managers are there to do something before and/or after a block of code. The obvious use case is a file open (where you want the file to always be closed when the block ends). But after the block ends, even if the file is closed, the file object still has value to the rest of the code. You can check its `filename` property, you list the file descriptor, test whether it's open or not, etc. But to do those things, the object has to still exist.